--- busybox-1.22.1/coreutils/uname.c.original	2014-12-14 17:15:12.373614915 +0000
+++ busybox-1.22.1/coreutils/uname.c	2014-12-14 17:17:38.718899645 +0000
@@ -137,8 +137,8 @@
 		strcpy(uname_info.name.machine, "sparc");
 	}
 #endif
-	strcpy(uname_info.processor, unknown_str);
-	strcpy(uname_info.platform, unknown_str);
+	strcpy(uname_info.processor, uname_info.name.machine);
+	strcpy(uname_info.platform, uname_info.name.machine);
 	strcpy(uname_info.os, "GNU/Linux");
 #if 0
 	/* Fedora does something like this */
