--- coreutils-8.22/src/uname.c.original	2014-12-14 17:06:36.309138237 +0000
+++ coreutils-8.22/src/uname.c	2014-12-14 17:07:57.061832947 +0000
@@ -259,6 +259,7 @@
 main (int argc, char **argv)
 {
   static char const unknown[] = "unknown";
+  static char const arm[] = "armv7l";
 
   /* Mask indicating which elements to print. */
   unsigned int toprint = 0;
@@ -337,6 +338,12 @@
 # endif
         }
 #endif
+
+      if (element == unknown)
+        {
+          element = arm;
+        }
+
       if (! (toprint == UINT_MAX && element == unknown))
         print_element (element);
     }
@@ -362,6 +369,12 @@
             element = hardware_platform;
         }
 #endif
+
+      if (element == unknown)
+        {
+          element = arm;
+        }
+
       if (! (toprint == UINT_MAX && element == unknown))
         print_element (element);
     }
